src/main/java/com/softwareverde/bitcoin/block/MutableBlock.java

```diff
@@ -24,7 +24,7 @@
     protected Long _timestamp;
     protected Difficulty _difficulty;
     protected Long _nonce;
-    protected MerkleTreeNode<Transaction> _merkleTree = new MerkleTreeNode<>();
+    final MerkleTreeNode<Transaction> _merkleTree = new MerkleTreeNode<>();
```

Comment:
There's no clear benefit to this change and the default access restricts visibility of this member to subclasses.  Unless there is a specific intent behind hiding the member from subclass (e.x. should be mutated via other internal methods only, like _addTransactionToMerkleTree) (and even then, the more proper access modifier would be `private`), `protected` should be the default non-private access modifier.

Reply:
>The default access restricts visibility of this member to subclasses. 

Not if the classes are declared within the same package.

I guess it's a style thing. I tend to keep things private unless they're really necessary.

If I haven't implemented a sub class, or a sub class hasn't needed to access a parent's property I don't need to leave access open and open myself to weird unexpected behaviors done by children.

Reply:
now, the `final` I suggest you keep it, you don't ever re-assign that `_merkleTree` and if a subclass needs to do it, it's probably a sign of bad design.

Reply:
I totally agree with a lot of your `final` changes.  Thanks for combing through and finding them; I appreciate that.

The protected vs private vs default thing gets hairy, and ultimately, I will never be happy because I do not agree with the way Java handles access modifiers.  I tend to trust sub-class-authors (that they've viewed the parent and understand its nuances), and distrust invokers of the class (i.e. make it as idiot-proof as possible ...future-me being the idiot).  Unfortunately with Java, protected also grants intra-package access, which I lament, but the default access is the opposite of what I'd like.  Ultimately, and after a lot of frustration and internal debate, the compromise we've come up with is to create far more packages than normal and then using protected while pretending it does what we actually want.  Therefore `private` becomes the "hey! don't mess with this!" modifier for those cases, and `protected` for subclass-friendly (if you really know what you're doing), and `public` for the obvious.

We are legitimately open to better stylistic decisions for how to better convey the intended purpose.  I secretly hope you'll have a secret java command that will give us all what we want. :)

Reply:
> the compromise we've come up with is to create far more packages than normal 

HA! I was wondering why so many packages :)

Gotcha, we all have this mantras depending on our experience, mine has been mostly Java and the mantras have been things like:

 - `private final` unless you need it now or in the future, encapsulate and minimize mutability at all cost to avoid unexpected behavior

 -  Favor composition over inheritance, embrace functional interfaces (which is easier if you avoid mutability)

I went over such changes because my IDE (IntelliJ) lights up to the point you can ignore it when you go over bitcoin-verde's code, I'm sure all these lint warnings are in most cases good to follow, after we started following them our apps became a lot more stable, very few crashes daily and we have millions of active users monthly on thousands of different android devices and android OS versions, along with desktop (Win, Mac, Linux)